### PR TITLE
[ORTModule] ATen Support for torch.nn.GroupNorm

### PR DESCRIPTION
**Pull request by iK1D**
Model [huggingface's diffusers library](https://github.com/huggingface/diffusers) has torch.nn.GroupNorm which will be exported to sub-graph containing ONNX's InstanceNormalization, which is lack of gradient. The implementation of ORT's InstanceNormalization will call cuDNN's BatchNorm for part of computation, which is not efficient compared to PyTorch's implementation. This PR is to use ATen fallback to support this torch module, including its forward and backward.

### CODE REVIEW

File: orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py [hunk 1]
```diff
@@ -663,3 +663,33 @@ def einsum_internal(g, equation, tensor_list):
...
+    from torch.onnx.symbolic_opset9 import group_norm as group_norm_generic
+
+    if weight is None or sym_help._is_none(weight) or bias is None or sym_help._is_none(bias):
+        return group_norm_generic(g, input, num_groups, weight, bias, eps, cudnn_enabled)
+
+    shape = g.op("Shape", input)
+    size = g.op("Size", input)
+    N = g.op("Gather", shape, g.op("Constant", value_t=torch.tensor(0, dtype=torch.long)), axis_i=0)
+    C = g.op("Gather", shape, g.op("Constant", value_t=torch.tensor(1, dtype=torch.long)), axis_i=0)
+    HxW = g.op("Div", size, g.op("Mul", N, C))
```
Comment (askhade) [inline orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py:682]: Can you explain N, C and HxW construction?

**Review by askhade — APPROVED** [review]
LGTM

File: orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py [hunk 2]
```diff
@@ -663,3 +663,33 @@ def einsum_internal(g, equation, tensor_list):
...
+def group_norm(g, input, num_groups, weight, bias, eps, cudnn_enabled):
+    # Torch's group_norm's weight and bias are optional, its gradient has a bool[3] augment to indicate
+    # whether to compute the gradient for input, weight, bias. For simplicity of the gradient graph builder,
+    # we support only the case that weight and bias are not None.
+    from torch.onnx.symbolic_opset9 import group_norm as group_norm_generic
+
+    if weight is None or sym_help._is_none(weight) or bias is None or sym_help._is_none(bias):
+        return group_norm_generic(g, input, num_groups, weight, bias, eps, cudnn_enabled)
+
+    shape = g.op("Shape", input)
```
Comment (baijumeswani) [inline orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py:678]: Can you share the link to the pytorch code that does this in going from `F.group_norm` to `native_group_norm`?

Comment (iK1D) [reply orttraining/orttraining/python/training/ortmodule/_custom_op_symbolic_registry.py:678]: https://github.com/pytorch/pytorch/blob/77d29bcee200f04bece4a86283acfb8e1ec830ad/aten/src/ATen/native/group_norm.cpp#L152

**Review by baijumeswani — APPROVED** [review]
